airbyte-ci/connectors_ci/metadata_service/orchestrator/orchestrator/sensors/catalog_sensors.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+from dagster import sensor, RunRequest, SkipReason, SensorDefinition, SensorEvaluationContext, build_resources, DefaultSensorStatus
+
+
+def generate_composite_etag_cursor(etags: List[str]):
+    return ":".join(etags)
+
+
+def catalog_updated_sensor(job, resources_def) -> SensorDefinition:
+    """
+    This sensor is responsible for checking if the latest catalog has been updated in GCS.
+    If it has, it will trigger the given job.
+    """
+
+    @sensor(
+        name=f"{job.name}_on_catalog_updated",
+        job=job,
+        minimum_interval_seconds=30,
+        default_status=DefaultSensorStatus.STOPPED,
+    )
+    def catalog_updated_sensor_definition(context: SensorEvaluationContext):
+        context.log.info("Starting gcs_catalog_updated_sensor")
+
+        with build_resources(resources_def) as resources:
+            context.log.info("Got resources for gcs_catalog_updated_sensor")
+
+            etag_cursor = context.cursor or None
+            context.log.info(f"Old etag cursor: {etag_cursor}")
+
+            new_etag_cursor = generate_composite_etag_cursor(
+                [resources.latest_oss_catalog_gcs_file.etag, resources.latest_cloud_catalog_gcs_file.etag]
```

Comment:
What kind of operation happens to query the etag on the GCP side? Do we make light calls that only provide the etag or are we reading the full file every 30s? I'm wondering if there could be an approach that would not require an actual file read with low "load" on GCP API so that we're not billed for file read but only for etag consumption.

Reply:
You can get ETAGs with just HEAD requests (and no body payload returned).

```
curl -v -X HEAD https://storage.googleapis.com/prod-airbyte-cloud-connector-metadata-service/oss_catalog.json
```

Reply:
Thanks Evan! Exactly this. We get to know if the file changed without downloading it. Pretty cool.

Reply:
Oh, and to complete the picture - the etag is calculated by GCS when the file is uploaded. Sort of equivalent to a MD5 hash of the content.